Refactor omero-prometheus-tools

manics opened the pull request:
Replaces https://github.com/IDR/omero-prometheus-tools/pull/2

- Update the omero session metrics with user and group counts
- Pip installable
- Login once on startup instead of for each request
- Exit if an error occurs so it's clear to the user (or systemd) that there is a problem
- Add travis (needs to be enabled on this repo)

Testing: `omero-prometheus-sessions.py -s omero.example.org -u public -w public`. See the updated top-level README.md

Tag: `0.1.0`

manics committed:
Add counts for number of OMERO users and groups

manics committed:
Uee omero_{users,groups}_total with labels for counts

manics committed:
Convert to pip installable package

manics committed:
Rewrite sessions systemd file

manics committed:
Update readme and dockerfile, add travis

manics committed:
Include prometheus-omero-sessions.service in python package

manics committed:
Refactor so login happens once at startup

This means this agent accurately reflects whether the serive is up, since it will only start serving metrics if the login succeeded

manics committed:
Remove sessions/README.md. Fix comment in setup.py

joshmoore commented on Dockerfile:
```diff
@@ -0,0 +1,18 @@
+    omego download python --sym OMERO.py && \
+    echo /opt/omero/OMERO.py/lib/python > /usr/lib/python2.7/site-packages/omero.pth
+
+RUN pip install https://github.com/manics/omero-prometheus-tools/archive/pip-package.zip
```
This seems odd. Shouldn't this be building the source in the current tree?

manics replied:
Could do

joshmoore reviewed (commented):
When this is pip-installed how is it used? The service files seem to assume it's under /opt/o-p-t

manics commented:
You can just pip install it and run it directly (see description). The service file is for convenience if you want to run it with systemd. Alternative is to get rid of it and create it in a playbook?

manics committed:
Add metric server for counts of omero objects

manics committed:
Add service for counts. Move all service files to service/

manics committed:
Build docker from current source.

Remove entrypoint from docker since there are now multiple scripts

manics committed:
Ensure omero-prometheus-counts.py uses a diff name for login time

manics committed:
Add Fileset to counts

manics committed:
Move service to etc

manics committed:
Rewrite counts to use configurable queries

manics committed:
remove labelsets which are no longer present

kennethgillen commented:
Where does the monitoring user's password sit on-disk, and with what filesystem-level permissions?

manics committed:
Remove service files

manics commented:
user/pass can be stored in a separate config file  or startup script managed elsewhere

manics commented:
Something to consider: worth merging these two separate `bin/` scripts into a single script?

manics committed:
Combine all metrics into a single script

manics committed:
Set the login_time metric in the top-level script

manics committed:
Default to port 9449

Added to https://github.com/prometheus/prometheus/wiki/Default-port-allocations

manics commented:
I've combined them, set a default port of 9449, added to https://github.com/prometheus/prometheus/wiki/Default-port-allocations

joshmoore reviewed (commented):
Generally looks good for a 0.1.0. A few questions:

 * I'm not sure which 2 scripts you're referring to.
 * Objections to migrating to github.com/ome?
 * Objections to holding off on the pypi package?

manics commented:
> I'm not sure which 2 scripts you're referring to.

Previously there were separate scripts for the two groups of metrics (sessions and object counts). I've combined them.

> Objections to migrating to github.com/ome?

:+1:

> Objections to holding off on the pypi package?

:+1: I'm happy if we never put this on pypi since it'll be deployed with a separate ansible role and it avoids another manual step in the release process

If you're happy can you tag and transfer?

joshmoore commented:
Done: https://github.com/ome/omero-prometheus-tools/releases/tag/0.1.0
